### PR TITLE
Disable msse2 flags on Apple Silicon

### PR DESCRIPTION
**Pull request by deniskoronchik**
# Issue

What issue is this PR targeting? If there is no issue that addresses the problem, please open a corresponding issue and link it here.

## Tasklist

 - [ ] Add tests
 - [ ] Add #fixes with the issue number that this PR addresses
 - [ ] Update the docs with any new request parameters or changes to behavior described
 - [ ] Update the [changelog](CHANGELOG.md)
 - [ ] If you made changes to the lua files, update the [taginfo](taginfo.json) too.

## Requirements / Relations

 Link any requirements here. Other pull requests this PR is based on?

### CODE REVIEW

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -108,7 +108,7 @@ function(valhalla_module)
   # These options do not affect the ABI and should therefore be used whenever possible for
   # predictable numerical results
   # For the x86-64 compiler, these extensions are enabled by default.
-  if(CMAKE_SYSTEM_PROCESSOR MATCHES "amd64|AMD64|x64|x86|x86_64")
+  if(CMAKE_SYSTEM_PROCESSOR MATCHES "amd64|AMD64|x64|x86|x86_64") AND NOT CMAKE_OSX_ARCHITECTURES MATCHES "arm64")
```
Comment (kevinkreiser) [inline src/CMakeLists.txt:111]: strange, that seems redundant to me. i guess on ios builds it advertizes itself as x86 through `CMAKE_SYSTEM_RPOCESSOR` variable?

Comment (deniskoronchik) [reply src/CMakeLists.txt]: `CMAKE_SYSTEM_RPOCESSOR  = x86_64` when we build it with arm64. I will figure out more soon, maybe some of dependencies override it

Comment (averkhaturau) [reply src/CMakeLists.txt]: @kevinkreiser , it fixes an issue with new Macs on arm64.

Comment (deniskoronchik) [reply src/CMakeLists.txt]: `CMAKE_OSX_ARCHITECTURES=arm64`  setup `-arch arm64` flag to compiler. I don't know, why `CMAKE_SYSTEM_PROCESSOR=x86_64`

Comment (deniskoronchik) [reply src/CMakeLists.txt]: https://gitlab.kitware.com/cmake/cmake/-/issues/20989

Comment (kevinkreiser) [reply src/CMakeLists.txt]: oh i see its for the new mac arm architectures, thanks

**Review by kevinkreiser — DISMISSED** [review]

**Review by averkhaturau — APPROVED** [review]

**Pull request merged**